decode_test.go

```diff
@@ -982,6 +982,18 @@ func (s *S) TestDecoderErrors(c *C) {
 	}
 }
 
+func (s *S) TestParserError(c *C) {
+	var v struct {
+		A, B int
+	}
+	data := "a: 1\n=\nb: 2"
+	err := yaml.Unmarshal([]byte(data), &v)
+	c.Assert(err, DeepEquals, &yaml.ParserError{
+		Message: "could not find expected ':'",
+		Line:    2,
+	})
+}
+
```

Comment:
would it be useful to add a test using `errors.As()` for inspecting the wrapped errors as a demonstration of the Unwrap []errors implementation?

Reply:
This is also done